src/main/java/rest/BlogController.java

```diff
@@ -0,0 +1,92 @@
+package rest;
+
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Controller
+public class BlogController {
+
+    @Autowired
+    private BlogServiceRepository blogRepository;
+
+    @RequestMapping({"/", "/blog"})
+    public String index(Model model) {
+        model.addAttribute("blogs", blogRepository.findAll());
+        return "index";
+    }
+
+    @RequestMapping("/blog/{id}")
+    public String show(Model model, @PathVariable String id) {
+        int blogId = Integer.parseInt(id);
+        List<Blog> blogs = new ArrayList<>();
+        blogs.add(blogRepository.findById(blogId).get());
+        model.addAttribute("blogs", blogs);
+        return "index";
+    }
+
+    @GetMapping("/blog/search")
+    public String search(Model model, @RequestParam(value = "text", required = false) String text) {
+        if (text != null) {
+            model.addAttribute("blogs", blogRepository.findByTitleContainingOrContentContaining(text, text));
```

Comment:
function name "findByTitleContainingOrContentContaining" should be renamed to "findByTitleOrContent"